tests/test_integration.py

```diff
@@ -1799,3 +1799,47 @@ async def test_suite_not_a_suite_directory(envsonschema, tmp_path):
         exit_code=-1,
     )
     assert re.search(r"does not contain .* cases", stderr), stderr
+
+
+@pytest.mark.asyncio
+async def test_dialect_warning_validate(envsonschema, tmp_path):
+    tmp_path.joinpath("schema.json").write_text(
+        '{"$schema": "https://json-schema.org/draft/2020-12/schema"}',
```

Comment:
Don't write JSON by hand, use `json.dumps` (and same below in the next test).

Reply:
Okay, I saw it in various places in the tests so thought that was the standard

Reply:
Got it, if you see it elsewhere in the file it's fine to leave then, I forget what we do